diff -up esmf/build/common.mk.orig esmf/build/common.mk
--- esmf/build/common.mk.orig	2020-01-31 16:18:34.913994609 +0100
+++ esmf/build/common.mk	2020-01-31 16:18:54.961994594 +0100
@@ -1380,7 +1380,7 @@ ESMF_F90LINKLIBS          += $(ESMF_NETC
 ESMF_F90LINKRPATHSTHIRD   += $(addprefix $(ESMF_F90RPATHPREFIX),$(subst -L,,$(filter -L%,$(ESMF_NETCDF_LIBS))))
 endif
 ifdef ESMF_NETCDF_LIBPATH
-ifdef $(ESMF_NETCDF_FPATH)
+ifdef ESMF_NETCDF_FPATH
 ifneq ($(ESMF_NETCDF_CPATH),$(ESMF_NETCDF_FPATH))
 ESMF_CXXLINKPATHSTHIRD    += -L$(ESMF_NETCDF_FLIBPATH)
 ESMF_F90LINKPATHSTHIRD    += -L$(ESMF_NETCDF_FLIBPATH)
