diff -up Cartopy-0.13.1/lib/cartopy/io/shapereader.py.orig Cartopy-0.13.1/lib/cartopy/io/shapereader.py
--- Cartopy-0.13.1/lib/cartopy/io/shapereader.py.orig	2021-09-02 17:01:12.256120098 +0200
+++ Cartopy-0.13.1/lib/cartopy/io/shapereader.py	2021-09-02 17:05:32.046345376 +0200
@@ -279,8 +279,8 @@ class NEShpDownloader(Downloader):
     # Define the NaturalEarth URL template. The natural earth website
     # returns a 302 status if accessing directly, so we use the naciscdn
     # URL directly.
-    _NE_URL_TEMPLATE = ('http://naciscdn.org/naturalearth/{resolution}'
-                        '/{category}/ne_{resolution}_{name}.zip')
+    _NE_URL_TEMPLATE = ('https://naturalearth.s3.amazonaws.com/'
+                        '{resolution}_{category}/ne_{resolution}_{name}.zip')
 
     def __init__(self,
                  url_template=_NE_URL_TEMPLATE,
