diff -up Metview-5.10.1-Source/metview/CMakeLists.txt.orig Metview-5.10.1-Source/metview/CMakeLists.txt
--- Metview-5.10.1-Source/metview/CMakeLists.txt.orig	2020-12-10 07:19:54.488924113 +0100
+++ Metview-5.10.1-Source/metview/CMakeLists.txt	2020-12-10 07:20:01.122880606 +0100
@@ -583,6 +583,9 @@ if( METVIEW_ODB )
         ${ODB_API_INCLUDE_DIRS}/../eclib
     )
 endif()
+if(CURL_FOUND)
+    list(APPEND METVIEW_STANDARD_INCLUDE_DIRS ${CURL_INCLUDE_DIRS})
+endif()
 
 
 set(STANDARD_METVIEW_LIBS Metview MvFTimeUtil) ### ${METVIEW_3RDPARTY_LIBS}"
