--- g2clib-1.6.2.p12/simunpack.c	2021-01-07 09:11:50.845682809 +0100
+++ g2clib-1.6.2.p123/simunpack.c	2021-01-07 09:23:07.786142031 +0100
@@ -51,7 +51,7 @@ g2int simunpack(unsigned char *cpack,g2i
       nbits = idrstmpl[3];
       itype = idrstmpl[4];
 
-      ifld=(g2int *)calloc(ndpts,sizeof(g2int));
+      ifld=(g2int *)calloc((ndpts ? ndpts : 1),sizeof(g2int));
       if ( ifld == 0 ) {
          fprintf(stderr,"Could not allocate space in simunpack.\n  Data field NOT upacked.\n");
          return(1);
